### ISSUE TITLE
Juniper Remediation Incorrect

### ISSUE THREAD

**Issue by jdrew82** [netdevops/hier_config#129]
After testing the Juniper remediation feature with the following options:

```
{
    "full_text_sub": [],
    "idempotent_commands": [],
    "idempotent_commands_blacklist": [],
    "indent_adjust": [],
    "negation": "delete",
    "negation_default_when": [],
    "negation_negate_with": [],
    "ordering": [],
    "parent_allows_duplicate_child": [],
    "per_line_sub": [],
    "sectional_exiting": [],
    "sectional_overwrite": [],
    "sectional_overwrite_no_negate": [],
    "style": "junos",
    "syntax_style": "juniper"
}
```

I found that the returned remediation simply appends the negation word to the start of the configuration lines instead of replacing the set keyword.

**Comment by aedwardstx** [comment]
This can be addressed this with either an idempotent_command rule e.g. https://github.com/netdevops/hier_config/blob/master/hier_config/platforms/hp_procurve/driver.py#L238 or by adding this functionality to the idempotent_for Driver method e.g. https://github.com/netdevops/hier_config/blob/master/hier_config/platforms/hp_procurve/driver.py#L115. I have a suspicion that if we were to add support for Junipers nested config syntax, this problem would be little easier to solve for.

**Comment by jtdub** [comment]
@jdrew82 -- Are you able to test this again? If you're still having issues, can you provide more details?

**Comment by jdrew82** [comment]
I don't think this is related to idempotent commands as in this case, they're not actually idempotent. I expect the word `set` at the start of the line with the word `delete`.
